imap/xapian_wrap.cpp

```diff
@@ -1110,6 +1119,7 @@ int add_text_part(xapian_dbw_t *dbw, const struct buf *part, int partnum)
         if (config_getswitch(IMAPOPT_SEARCH_INDEX_LANGUAGE)){
             // Index by language.
 #ifndef HAVE_CLD2
+            // XXX is this really an "IOERROR"?
             syslog(LOG_ERR, "IOERROR: Xapian: language indexing requires CLD2 library");
             return IMAP_IOERROR;
 #else
```

Comment:
@rsto what do you think about this?

I feel like this sort of "we weren't compiled to support this config option" error would usually be a `fatal()` rather than a syslog-and-return-error.  But, we're in C++ code here, and maybe we deliberately don't `fatal()` because it's unsafe?  So I think syslog-and-return-error is correct, but thought I'd double-check...

Reply:
We have a write-opened Xapian database open at this stage, so I chose to abort gracefully. Also: this codepath is called in two codepaths mostly:
1. squatter during indexing: in which case the return error should bubble up all to squatter and the process ends with an error code
2. sieve scripts which use the jmap extensions, in which case it should better be handled in sieve how to handle that message (e.g. just put in the inbox).

Reply:
Great, that makes sense, thanks for clarifying. :)